iam-token-manager/v1.ts

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2015 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import extend = require('extend');
+import { sendRequest } from '../lib/requestwrapper';
+
+export type Options = {
+  iamApikey?: string;
+  iamAccessToken?: string;
+  iamUrl?: string;
+}
+
+export interface IamTokenData {
+  access_token: string;
+  refresh_token: string;
+  token_type: string;
+  expires_in: number;
+  expiration: number;
+}
```

Comment:
I suspect you've made a conscious decision. But it seems a little odd that `Options` uses `camelCase` while `IamTokenData` uses `snake_case`.

Reply:
A conscious decision, yes, because the IAM service response object had exactly those keys. It is designed to match that. I should maybe leave a comment because you aren't the first person to bring it up!